core/src/component/registry.rs

```diff
@@ -50,16 +50,18 @@ where
         // Ensure all component names are unique
         let mut names = HashSet::new();
 
-        for component in &result.components {
+        result.sort();
```

Comment:
It seems like having this here means the components won't actually get sorted?

Reply:
Are you sure?

The intention was to do the sort (an in-place operation that mutates `result`) before iterating over the contents of `result`, checking uniqueness and then pushing qualifying elements into the registry.

Reply:
Upon a second look, this seems ok.